sdk/src/settings/mod.rs

```diff
@@ -160,67 +176,123 @@ impl SettingsValidate for Trust {
+    /// <https://spec.c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#_bmff_based_hash>
+    ///
+    /// [`BmffHash`]: crate::assertions::BmffHash
+    pub merkle_tree_max_proofs: usize,
+    /// Maximum amount of data in megabytes that will be loaded into memory before
+    /// it's stored in temporary files on the disk.
+    ///
+    /// This option defaults to 512MB and can result in noticeable performance improvements.
+    pub backing_store_memory_threshold_in_mb: usize,
 }
 
 impl Default for Core {
     fn default() -> Self {
         Self {
-            debug: false,
-            hash_alg: "sha256".into(),
-            soft_hash_alg: None,
-            salt_jumbf_boxes: true,
-            prefer_box_hash: false,
             merkle_tree_chunk_size_in_kb: None,
             merkle_tree_max_proofs: 5,
-            compress_manifests: true,
-            max_memory_usage: None,
             backing_store_memory_threshold_in_mb: 512,
-            // prefer_update_manifests: true,
         }
     }
 }
 
 impl SettingsValidate for Core {
     fn validate(&self) -> Result<()> {
-        match self.hash_alg.as_str() {
-            "sha256" | "sha384" | "sha512" => Ok(()),
-            _ => Err(Error::UnsupportedType),
-        }
+        Ok(())
     }
 }
 
-// Settings for verification options
+/// Settings to configure the verification process.
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-#[allow(unused)]
-pub(crate) struct Verify {
-    verify_after_reading: bool,
-    verify_after_sign: bool,
-    verify_trust: bool,
-    verify_timestamp_trust: bool,
-    ocsp_fetch: bool,
-    remote_manifest_fetch: bool,
-    check_ingredient_trust: bool,
-    skip_ingredient_conflict_resolution: bool,
-    strict_v1_validation: bool,
+pub struct Verify {
+    /// Whether to verify the manifest after reading in the [`Reader`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Disabling validation can improve reading performance, BUT it carries the risk of reading an invalid
+    /// manifest.
+    /// </div>
+    ///
+    /// [`Reader`]: crate::Reader
+    pub verify_after_reading: bool,
+    /// Whether to verify the manifest after signing in the [`Builder`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Disabling validation can improve signing performance, BUT it carries the risk of signing an invalid
+    /// manifest.
+    /// </div>
+    ///
+    /// [`Builder`]: crate::Builder
+    pub verify_after_sign: bool,
+    /// Whether to verify certificates against the trust lists specified in [`Trust`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Verifying trust is REQUIRED by the C2PA spec. This option should only be used for development or testing.
+    /// </div>
+    pub verify_trust: bool,
```

Comment:
Maybe we should have `verify_c2pa_trust` and `verify_cawg_trust` to be clear they are separate and also to clarify it doesn't encompass timestamp trust. I'll add to the docs that `verify_timestamp_trust` should be used in that case.

Reply:
It seems to me that if you configure the Trust settings, you probably want to use them. So why do we need a separate set of flags to enable them?